src/settings/target/site.rs

```diff
@@ -1,43 +1,64 @@
+use std::env;
+use std::fs;
 use std::path::PathBuf;
 
 use serde::{Deserialize, Serialize};
 
+use crate::commands::generate::run_generate;
+
 const SITE_ENTRY_POINT: &str = "workers-site";
 
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Site {
-    pub bucket: String,
+    pub bucket: PathBuf,
     #[serde(rename = "entry-point")]
-    pub entry_point: Option<String>,
+    entry_point: Option<PathBuf>,
     pub include: Option<Vec<String>>,
     pub exclude: Option<Vec<String>>,
 }
 
 impl Site {
     pub fn new(bucket: &str) -> Site {
         let mut site = Site::default();
-        site.bucket = String::from(bucket);
+        site.bucket = PathBuf::from(bucket);
 
         site
     }
 
     // if the user has configured `site.entry-point`, use that
     // as the build directory. Otherwise use the default const
     // SITE_ENTRY_POINT
-    pub fn build_dir(&self, current_dir: PathBuf) -> Result<PathBuf, std::io::Error> {
+    pub fn entry_point(&self) -> Result<PathBuf, std::io::Error> {
+        let current_dir = env::current_dir()?;
```

Comment:
eventually methods like these that determine where a file should be found relative to the project root should probably take as an argument the actual project root path, rather than the current directory. for the moment, everything runs relative to root because that is where wrangler.toml is and we assume you are running wrangler adjacent to the toml.

Reply:
which makes me want to write a refactor issue

Reply:
^ this would also be a step towards better monorepo support!

Reply:
and/or just being able to run wrangler from anywhere inside a project. i'll write the issue and merge this.

Reply:
https://github.com/cloudflare/wrangler/issues/860